universal-application-tool-0.0.1/app/controllers/LoginController.java

```diff
@@ -0,0 +1,53 @@
+package controllers;
+
+import java.util.Optional;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Named;
+import org.pac4j.core.context.session.SessionStore;
+import org.pac4j.core.exception.http.RedirectionAction;
+import org.pac4j.core.http.adapter.HttpActionAdapter;
+import org.pac4j.oidc.client.OidcClient;
+import org.pac4j.play.PlayWebContext;
+import org.pac4j.play.http.PlayHttpActionAdapter;
+import play.mvc.Controller;
+import play.mvc.Http;
+import play.mvc.Result;
+
+/**
+ * These routes can be hit even if you are logged in already, which is what allows the merge logic -
+ * normally you need to be logged out in order to be redirected to a login page.
+ */
+public class LoginController extends Controller {
+  @Inject
+  @Nullable
+  @Named("idcs")
+  OidcClient idcsClient;
+
+  @Inject
+  @Nullable
```

Comment:
Question (blocking): the case where this and/or idcsClient are null isn't handled here, why are they nullable?

Reply:
Handled that case, but the answer is because guice complains about injecting null into something that isn't annotated nullable, but this is only true if you don't provide secrets, so the unhandled crash wasn't a big deal.